src/main/java/hudson/plugins/jira/JiraCreateIssueNotifier.java

```diff
@@ -96,7 +98,7 @@ public BuildStepMonitor getRequiredMonitorService() {
     @Override
     public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListener listener) throws IOException, InterruptedException {
 
-        String jobDirPath = Jenkins.getInstance().getBuildDirFor(build.getProject()).getPath();
+        String jobDirPath = build.getProject().getBuildDir().getPath();
```

Comment:
@artkoshelev can you take a look at this? I started some refactoring but probably could use some help.
E.g. I'm not sure if this change I did is good, still trying to dig through all of this undocumented API... Any remarks about the changes or test optimizations are welcome.